src/main/webapp/app/exercises/shared/dashboards/tutor/exercise-assessment-dashboard.component.ts

```diff
@@ -598,47 +598,63 @@ export class ExerciseAssessmentDashboardComponent implements OnInit {
-        if (!this.exercise || !this.exercise.type || !submission) {
-            return;
-        }
-
-        this.openingAssessmentEditorForNewSubmission = true;
-        const submissionId: number | 'new' = submission === 'new' ? 'new' : submission.id!;
+    getAssessmentLink(submission: Submission | 'new'): string[] {
+        const submissionUrlParameter: number | 'new' = submission === 'new' ? 'new' : submission.id!;
         let participationId = undefined;
         if (submission !== 'new' && submission.participation !== undefined) {
             participationId = submission.participation!.id;
         }
-        const url = getLinkToSubmissionAssessment(this.exercise.type!, this.courseId, this.exerciseId, participationId, submissionId, this.examId, this.exerciseGroupId);
+        return getLinkToSubmissionAssessment(this.exercise.type!, this.courseId!, this.exerciseId, participationId, submissionUrlParameter, this.examId, this.exerciseGroupId);
+    }
+
+    /**
+     * Generates and returns the query parameters required for opening the assessment editor
+     * @param correctionRound
+     */
+    getAssessmentQueryParams(correctionRound = 0): object {
         if (this.isTestRun) {
-            this.router.navigate(url, { queryParams: { testRun: this.isTestRun, 'correction-round': correctionRound } });
+            return {
+                testRun: this.isTestRun,
+                'correction-round': correctionRound,
+            };
         } else {
-            this.router.navigate(url, { queryParams: { 'correction-round': correctionRound } });
+            return { 'correction-round': correctionRound };
         }
-        this.openingAssessmentEditorForNewSubmission = false;
     }
 
     /**
-     * Show complaint depending on the exercise type
-     * @param complaint that we want to show
+     * Generates and returns the query parameters required for opening a complaint
+     * @param complaint
      */
-    viewComplaint(complaint: Complaint) {
+    getComplaintQueryParams(complaint: Complaint) {
         const submission: Submission = complaint.result?.submission!;
         // numberOfAssessmentsOfCorrectionRounds size is the number of correction rounds
         if (complaint.complaintType === ComplaintType.MORE_FEEDBACK) {
-            this.openAssessmentEditor(submission, this.numberOfAssessmentsOfCorrectionRounds.length - 1);
+            return this.getAssessmentQueryParams(this.numberOfAssessmentsOfCorrectionRounds.length - 1);
+        }
+        const result = this.getSubmissionToViewFromComplaintSubmission(submission);
+
+        if (result !== undefined) {
+            return this.getAssessmentQueryParams(result.results!.length - 1);
         }
+    }
+
+    /**
+     * Returns either the corresponding submission to view or undefined.
+     * This complaint has to be a true complaint or else issues can arise.
+     * @param submission
+     */
+    getSubmissionToViewFromComplaintSubmission(submission: Submission) {
```

Comment:
```suggestion
    getSubmissionToViewFromComplaintSubmission(submission: Submission): Submission | undefined {
```
I'd prefer writing out the return type